packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -698,13 +697,44 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
       },
     );
   }
+
+  /// Builds [TextSpan] from current editing value.
+  ///
+  /// By default makes text in composing range appear as underlined.
+  /// Descendants can override this method to customize appearance of text.
+  TextSpan buildTextSpan() {
+    final int obscureShowCharacterAtIndex =
+        _obscureShowCharTicksPending > 0 ? _obscureLatestCharIndex : null;
+
+    if (!widget.obscureText && _value.composing.isValid) {
+      final TextStyle composingStyle = widget.style
+          .merge(const TextStyle(decoration: TextDecoration.underline));
+
+      return new TextSpan(style: widget.style, children: <TextSpan>[
+        new TextSpan(text: _value.composing.textBefore(_value.text)),
+        new TextSpan(
+            style: composingStyle,
+            text: _value.composing.textInside(_value.text)),
+        new TextSpan(text: _value.composing.textAfter(_value.text))
+      ]);
+    }
+
+    String text = _value.text;
+    if (widget.obscureText) {
+      text = RenderEditable.obscuringCharacter * text.length;
+      final int o = obscureShowCharacterAtIndex;
```

Comment:
you can inline the expression for obscureShowCharacterAtIndex here, rather than computing it early and putting it here